src/utils/mempool.rs

```diff
@@ -49,15 +80,30 @@ impl DoubleMemPool {
 ///
 /// This wrapper handles for you the creation of the tempfile and its synchronisation
 /// with the protocol.
+///
+/// Mempool internally tracks the lifetime of all buffers created from it and to ensure that
+/// this buffer count is correct all buffers must be attached to a surface. Once a buffer is attached to
+/// a surface it must be immediately commited to that surface before another buffer is attached.
+///
+/// Mempool will also handle the destruction of buffers and as such the `destroy()` method should not
+/// be used on buffers created from Mempool.
+///
+/// Mempool requires an implementation that will be called when the pool becomes free, this
+/// happens when all the pools buffers are released by the server.
```

Comment:
Please add something like "Overwriting the contents of the memory pool before it is completely freed may cause graphical glitches as it may corrupt the data while the compositor is reading it."